fastapi_websocket_pubsub/event_broadcaster.py

```diff
@@ -29,14 +29,81 @@ class BroadcasterAlreadyStarted(EventBroadcasterException):
     pass
 
 
+class EventBroadcasterContextManger:
+    """
+    Manages the context for the EventBroadcaster
+    Friend-like class of EventBroadcaster (accessing "protected" members )
+    """
+
+    def __init__(self, event_broadcaster: "EventBroadcaster", listen: bool = True, share: bool = True) -> None:
+        """
+        Provide a context manager for an EventBroadcaster, managing if it listens to events coming from the broadcaster 
+        and if it subscribes to the internal notifier to share its events with the broadcaster
+
+        Args:
+            event_broadcaster (EventBroadcaster): the broadcaster we manage the context for.
+            share (bool, optional): Should we share events with the broadcaster. Defaults to True.
+            listen (bool, optional): Should we listen for incoming events from the broadcaster. Defaults to True.
+        """
+        self._event_broadcaster = event_broadcaster
+        self._share: bool = share
+        self._listen: bool = listen
+        self._lock = asyncio.Lock()
+
+    async def __aenter__(self):
+        async with self._lock:
+            if self._listen:
+                self._event_broadcaster._listen_count += 1
+                if self._event_broadcaster._listen_count == 1:
+                    # We have our first listener start the read-task for it (And all those who'd follow)
+                    logger.info("Listening for incoming events from broadcaster (first listener started)")
+                    # Start task listening on incoming broadcasts
+                    self._event_broadcaster.start_reader_task()
+
+            if self._share:
+                self._event_broadcaster._share_count += 1
+                if self._event_broadcaster._share_count == 1:
+                    # We have our first publisher
+                    # Init the broadcast used for sharing (reading has its own)
+                    self._event_broadcaster._acquire_publishing_broadcast()                
+                    logger.info("Subscribing to ALL TOPICS, to share with broadcast channel")
```

Comment:
how about:
Subscribing to ALL TOPICS, and sharing messages with broadcast channel

Reply:
Sure ...